File: src/test/testdata/inspections/unsafeIterators/IntIntMap.java
package main;

import com.badlogic.gdx.utils.IntIntMap;

import java.util.Iterator;

class IntIntMapTest {

  private static IntIntMap map1 = new IntIntMap();

  private IntIntMap map2 = new IntIntMap(map1);

  static void test() {
    for (IntIntMap.Entry e: <warning>map1</warning>) {
      map1.remove(e.key, e.value);
    }
  }

  void foo() {


    IntIntMap.Entries e = <warning>map1.entries()</warning>;
    Iterator i = e.iterator();
    <warning>map2.keys()</warning>.toArray();
    IntIntMap.Values i2 = <warning>map2.values()</warning>;

    for (IntIntMap.Entry entry: <warning>map2.entries()</warning>) {
      for (int v: new IntIntMap.Keys(map1).toArray().toArray()) {
        new IntIntMap.Values(map2);
      }
    }

    new IntIntMap(map2).entries();
  }

}

File: src/test/testdata/inspections/unsafeIterators/IntIntMap.kt
package main

import com.badlogic.gdx.utils.IntIntMap


@Suppress("UNUSED_VARIABLE", "UNCHECKED_CAST")
internal class IntIntMapTestKotlin {

  private val map2 = IntIntMap(map1)

  fun foo() {


    val e = <warning>map1.entries()</warning>
    val i = e.iterator()
    <warning>map2.keys()</warning>.toArray()
    val i2 = <warning>map2.values()</warning>

    for (entry in <warning>map2.entries()</warning>) {
      for (v in IntIntMap.Keys(map1).toArray().toArray()) {
        IntIntMap.Values(map2)
      }
    }

    <warning>IntIntMap(map2).entries()</warning>

    val a = <warning>(if (map2.size > 1) map1 else IntIntMap()).keys()</warning>
    val b = IntIntMap.Entries(map1)
  }

  companion object {

    private val map1 = IntIntMap()

    fun test() {
      for (e in <warning>map1</warning>) {
        map1.remove(e.key, e.value)
      }
    }
  }



}
